### PR TITLE
Updates modal to allow for updates prop changes

### PR DESCRIPTION
**Pull request by kierangillen**
- We were seeing issues where new props were not correctly being rendered. This PR updates the Modal to always pass new body content as props unless the `refreshModalContentKey ` updates.
- This PR also removes an issue where the Modal fade in animation was firing twice due to the `refreshModalContent` `useEffect` call firing on render. This removes that function and only calls the fade when `refreshModalContentKey` updates.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit ff3c6eba9e4038605cada77e3e673bb6d4a54785

https://deploy-preview-554--artsy-palette.netlify.com

File: packages/palette/src/elements/Modal/Modal.tsx [hunk 1]
```diff
@@ -4,15 +4,18 @@ import styled from "styled-components"
...
 import { Serif } from "../Typography"
 
+/**
+ * refreshModalContentKey should change if the modal displays new content and should fade in/fade out with content update
+ */
 interface ModalProps {
   FixedButton?: JSX.Element
-  // refreshModalContent should change if the modal displays new content
-  refreshModalContent?: string
+  refreshModalContentKey?: string
```
Comment (damassi) [inline packages/palette/src/elements/Modal/Modal.tsx]: If this doesn't change and props are passed, does the content update immediately?

File: packages/palette/src/elements/Modal/Modal.tsx [hunk 2]
```diff
@@ -96,7 +112,7 @@ export const Modal: SFC<ModalProps> = ({
     }
   }, [show])
 
-  const updateVisibleContent = () => {
+  const fadeInNewContent = () => {
```
Comment (damassi) [inline packages/palette/src/elements/Modal/Modal.tsx]: 👍

File: packages/palette/src/elements/Modal/Modal.tsx [hunk 3]
```diff
@@ -38,7 +41,7 @@ const AnimatedView = animated(Box)
  */
 export const Modal: SFC<ModalProps> = ({
   children,
-  refreshModalContent,
+  refreshModalContentKey,
```
Comment (erikdstock) [inline packages/palette/src/elements/Modal/Modal.tsx]: :+1:

File: packages/palette/src/utils/usePrevious.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+/**
+ * usePrevious
+ * Stores state or props in a ref to compare against new props or state
+ */
+export function usePrevious(value) {
+  const ref = useRef(value)
+  useEffect(() => {
+    ref.current = value
+  }, [value])
+  return ref.current
```
Comment (erikdstock) [inline packages/palette/src/utils/usePrevious.ts]: wow this is really cool and I'd like to understand more about how it works!

File: packages/palette/src/elements/Modal/Modal.tsx (hunk 1 above)
Comment (kierangillen) [reply packages/palette/src/elements/Modal/Modal.tsx]: yes it updates immediately if this key isn't changed.

Comment (damassi) [reply packages/palette/src/elements/Modal/Modal.tsx]: Nice, that's ideal 👍

Comment (damassi) [reply packages/palette/src/elements/Modal/Modal.tsx]: Can a comment be added somewhere describing this?

File: packages/palette/src/elements/Modal/Modal.tsx [hunk 5]
```diff
@@ -4,15 +4,18 @@ import styled from "styled-components"
...
 import { CloseIcon } from "../../svgs"
 import { ArtsyLogoBlackIcon } from "../../svgs/ArtsyLogoBlackIcon"
+import { usePrevious } from "../../utils/usePrevious"
 import { Box } from "../Box"
 import { Flex } from "../Flex"
 import { Spacer } from "../Spacer"
 import { Serif } from "../Typography"
 
+/**
+ * refreshModalContentKey should change if the modal displays new content and should fade in/fade out with content update
```
Comment (kierangillen) [inline packages/palette/src/elements/Modal/Modal.tsx]: @damassi there's a comment here describing it.

Comment (damassi) [reply packages/palette/src/elements/Modal/Modal.tsx]: > refreshModalContentKey should change if the modal displays new content and should fade in/fade out with content update. If `refreshModalContentKey` does not change, the content updates immedately.

File: packages/palette-docs/content/docs/elements/dialogs/Modal.mdx [hunk 6]
```diff
@@ -24,6 +33,7 @@ name: Modal
               width="100%"
               onClick={() =>
                 setState({
+                  contentKey: "456",
```
Comment (damassi) [inline packages/palette-docs/content/docs/elements/dialogs/Modal.mdx]: Can one more example be added that shows the content in the modal updating immediately, without fade in / fade out?

File: packages/palette/src/elements/Modal/Modal.tsx (hunk 5 above)
Comment (kierangillen) [reply packages/palette/src/elements/Modal/Modal.tsx]: Ah I see. Updating now.

File: packages/palette-docs/content/docs/elements/dialogs/Modal.mdx (hunk 6 above)
Comment (kierangillen) [reply packages/palette-docs/content/docs/elements/dialogs/Modal.mdx]: Just added

File: packages/palette-docs/content/docs/elements/dialogs/Modal.mdx [hunk 7]
```diff
@@ -9,6 +9,10 @@ name: Modal
   animation. See example by pressing the `Change content` button in this
   example.
 
+## Modal with refreshModalContentKey
```
Comment (damassi) [inline packages/palette-docs/content/docs/elements/dialogs/Modal.mdx]: > Modal with animated transitions
> ...

> Modal without animationed transitionos
> ...

**Comment by damassi** [comment]
Looks great 👍 

#mergeOnGreen

**Comment by artsyit** [comment]
:rocket: PR was released in v4.14.40 :rocket:

**Pull request merged**